src/Cake.NuGet.Tests/Unit/NuGetContentResolverTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Cake.NuGet.Tests.Fixtures;
+using Cake.Testing;
+using Xunit;
+
+namespace Cake.NuGet.Tests.Unit
+{
+    public sealed class NuGetContentResolverTests
+    {
+        public sealed class Tools
+        {
+            [Fact]
+            public void Should_Return_Exe_Files_By_Default()
+            {
+                // Given
+                var fixture = new NuGetToolContentResolverFixture("nuget:?package=Foo");
+                fixture.FileSystem.CreateFile("/Working/tools/Foo/Foo.exe");
+                fixture.FileSystem.CreateFile("/Working/tools/Foo/Foo.pdb");
+                fixture.FileSystem.CreateFile("/Working/tools/Foo/Foo.XML");
+
+                // When
+                var files = fixture.GetFiles();
+
+                // Then
+                Assert.Equal(1, files.Count);
+                Assert.Equal("/Working/tools/Foo/Foo.exe", files.First().Path.FullPath);
+            }
+
+            [Fact]
+            public void Should_Return_Dll_Files_By_Default()
+            {
+                // Given
+                var fixture = new NuGetToolContentResolverFixture("nuget:?package=Foo");
+                fixture.FileSystem.CreateFile("/Working/tools/Foo/Foo.dll");
+                fixture.FileSystem.CreateFile("/Working/tools/Foo/Foo.pdb");
+                fixture.FileSystem.CreateFile("/Working/tools/Foo/Foo.XML");
+
+                // When
+                var files = fixture.GetFiles();
+
+                // Then
+                Assert.Equal(1, files.Count);
+                Assert.Equal("/Working/tools/Foo/Foo.dll", files.First().Path.FullPath);
+            }
+
+            [Theory]
+            [InlineData("nuget:?package=Foo&include=/**/*.XML")]
+            [InlineData("nuget:?package=Foo&include=**/*.XML")]
+            [InlineData("nuget:?package=Foo&include=./**/*.XML")]
```

Comment:
The include/exclude is case sensitive due to the globber. How should we proceed?